test/jdk/java/awt/dnd/DragToAnotherScreenTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2005, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Frame;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Label;
+import java.awt.Toolkit;
+import java.awt.Window;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.StringSelection;
+import java.awt.datatransfer.Transferable;
+import java.awt.dnd.DnDConstants;
+import java.awt.dnd.DragGestureListener;
+import java.awt.dnd.DragSource;
+import java.awt.dnd.DropTarget;
+import java.awt.dnd.DropTargetAdapter;
+import java.awt.dnd.DropTargetDropEvent;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 6179157
+ * @summary Tests dnd to another screen
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual DragToAnotherScreenTest
+ */
+
+public class DragToAnotherScreenTest {
+    private static Label label0;
+    private static Label label1;
+    private static final int HGAP = 20;
+
+    private static final String INSTRUCTIONS = """
+                The following test is applicable for Single as well
+                as Multi-monitor screens.
+
+                If on multi-monitor screens then please position
+                the drag and drop windows on different screens.
```

Comment:
This can be done automatically… I mean `PassFailJFrame` won't be able to handle positioning of the second frame on another screen. Yet you can do it in the code.

Reply:
I wanted to add a comment that these enhancements could be implemented later because *“they're out of scope”* of converting applet tests.

At the same, it could be easier to do it while working on a test; returning to it later would require additional time to understand what needs to be done. But I agree additional features add up and eat time.

So, submitting bugs for additional improvements is a good balance. Otherwise, no one will ever remember.